Translate Typechecking with Proptypes page (#17)
reactjs/pt-BR.reactjs.org: content/docs/typechecking-with-proptypes.md

---
id: typechecking-with-proptypes

> title: Checagem de tipos com PropTypes

Reviewer: can we keep "Typechecking" in the translated version? it seems a very common term. something like the below, for example

```
Checagem de tipos (Typechecking) com PropTypes
```

permalink: docs/typechecking-with-proptypes.html
redirect_from:
  - "docs/react-api.html#typechecking-with-proptypes"
---

> Note:
> 

> > `React.PropTypes` foi movido para um package diferente desde a versão 15.5 do React. Por favor, use [a biblioteca `prop-types`](https://www.npmjs.com/package/prop-types).

Reviewer 2: ```suggestion
> `React.PropTypes` foi movido para um pacote diferente desde a versão 15.5 do React. Por favor, use [a biblioteca `prop-types`](https://www.npmjs.com/package/prop-types).
```

As suggested by the Style Guide https://github.com/reactjs/pt-BR.reactjs.org/issues/4

> > `React.PropTypes` foi movido para um package diferente desde a versão 15.5 do React. Por favor, use [a biblioteca `prop-types`](https://www.npmjs.com/package/prop-types).

Reviewer 3: ```suggestion
> `React.PropTypes` foi movido para um pacote diferente desde a versão 15.5 do React. Por favor, use [a biblioteca `prop-types`](https://www.npmjs.com/package/prop-types).
```

>
> Nós fornecemos [um script de codemod](/react/blog/2017/04/07/react-v15.5.0.html#migrating-from-react.proptypes) para automatizar a conversão.

> Na medida em que sua aplicação cresce, você pode capturar muitos bugs com checagem de tipos. Em algumas aplicações, você pode usar extensões do JavaScript como [Flow](https://flowtype.org/) ou [TypeScript](https://www.typescriptlang.org/) para checar os tipos de toda a sua aplicação. Porém, mesmo se você não usá-las, React possui algumas habilidades de checagem de tipos embutidas. Para executar a checagem de tipos nas props para um componente, você pode atribuir à propriedade em especial `propTypes`:

Reviewer 2: ```suggestion
Na medida em que sua aplicação cresce, você pode capturar muitos bugs com checagem de tipos. Em algumas aplicações, você pode usar extensões do JavaScript como [Flow](https://flowtype.org/) ou [TypeScript](https://www.typescriptlang.org/) para checar os tipos de toda a sua aplicação. Porém, mesmo se você não usá-las, React possui algumas habilidades de checagem de tipos nativas. Para executar a checagem de tipos nas props para um componente, você pode atribuir à propriedade em especial `propTypes`:
```

Suggestion based on another Style Guide, but I think it's a good one: https://github.com/cezaraugusto/You-Dont-Know-JS/blob/portuguese-translation/WORDREFERENCE.md#jarg%C3%B5es-de-programa%C3%A7%C3%A3o

Reviewer 4: This raised me a doubt, which probably will make me change my own translation, hahaha... when the text say **app** should we consider "aplicação" or "aplicativo"?

I think "aplicação" is more general, what do you think?

> Na medida em que sua aplicação cresce, você pode capturar muitos bugs com checagem de tipos. Em algumas aplicações, você pode usar extensões do JavaScript como [Flow](https://flowtype.org/) ou [TypeScript](https://www.typescriptlang.org/) para checar os tipos de toda a sua aplicação. Porém, mesmo se você não usá-las, React possui algumas habilidades de checagem de tipos embutidas. Para executar a checagem de tipos nas props para um componente, você pode atribuir à propriedade em especial `propTypes`:

Author: the verb `atribuir` asks for a complement (`bitransitivo`). Then, if we have `atribuir` we need a target "object". That's why I added the "crase" to it (artigo + preposição).

In `em especial`, I understood that `special propTypes property` mean a particular property, more to the translation of  `very one`" than to the specific word `special`.

> Na medida em que sua aplicação cresce, você pode capturar muitos bugs com checagem de tipos. Em algumas aplicações, você pode usar extensões do JavaScript como [Flow](https://flowtype.org/) ou [TypeScript](https://www.typescriptlang.org/) para checar os tipos de toda a sua aplicação. Porém, mesmo se você não usá-las, React possui algumas habilidades de checagem de tipos embutidas. Para executar a checagem de tipos nas props para um componente, você pode atribuir à propriedade em especial `propTypes`:

Author: but is really a trick thing. Let me know if you have another POV

};
```

> ### Valores Default de Props {#default-prop-values}

Reviewer: what if we keep with both translated/original versions here? 

```diff
-Valores Default de Props
+Valores Padrão de Props (Default Prop Values)
```

Author: got it. I was trying to figure out a good translation for `default` and couldn't find one. I kept the `default` word but as you mentioned, translating and putting on parenthesis seem better. When using parenthesis on Crowdin I got a warning

> Você pode definir valores default para suas `props` através da atribuição à propriedade especial `defaultProps`:

Reviewer: ```diff
-Você pode definir valores default
+Você pode definir valores padrão (default)
```

);
```

> Se você está usando um plugin Babel como [plugin-proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe React.Component. Essa sintaxe ainda não foi finalizada e irá exigir uma etapa de compilação para funcionar em um navegador. Para mais informações, veja [proposal-class-fields](https://github.com/tc39/proposal-class-fields).

Reviewer 2: Why the link was changed? I think it's better to keep the original one.

Author: the link ` https://babeljs.io/docs/plugins/transform-class-properties/` redirects to the `https://babeljs.io/docs/en/babel-plugin-proposal-class-properties` so I thought that would be better to change to the new one

> Se você está usando um plugin Babel como [plugin-proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe React.Component. Essa sintaxe ainda não foi finalizada e irá exigir uma etapa de compilação para funcionar em um navegador. Para mais informações, veja [proposal-class-fields](https://github.com/tc39/proposal-class-fields).

Reviewer 2: Also, I would change this bit to this:
```
você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe de componente React.
```

> Se você está usando um plugin Babel como [plugin-proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe de componente React. Essa sintaxe ainda não foi finalizada e irá exigir uma etapa de compilação para funcionar em um navegador. Para mais informações, veja [proposal-class-fields](https://github.com/tc39/proposal-class-fields).

Reviewer: I think it's better to keep the original proposal name `proposal-class-fields`

Author: where in this line?

> Se você está usando um plugin Babel como [plugin-proposal-class-properties](https://babeljs.io/docs/en/babel-plugin-proposal-class-properties), você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe de componente React. Essa sintaxe ainda não foi finalizada e irá exigir uma etapa de compilação para funcionar em um navegador. Para mais informações, veja [proposal-class-fields](https://github.com/tc39/proposal-class-fields).

Reviewer 5: ```suggestion
Se você está usando um plugin Babel como [plugin-proposal-class-properties](https://babeljs.io/docs/plugins/transform-class-properties), você também poderá declarar `defaultProps` como propriedade estática dentro de uma classe de componente React. Essa sintaxe ainda não foi finalizada e irá exigir uma etapa de compilação para funcionar em um navegador. Para mais informações, veja [proposal-class-fields](https://github.com/tc39/proposal-class-fields).
```

Author: the url `https://babeljs.io/docs/plugins/transform-class-properties` automatically redirects to `https://babeljs.io/docs/en/babel-plugin-proposal-class-properties`

Reviewer 5: Ok

}
```

> O `defaultProps` será usado para garantir que `this.props.name` tenha um valor se não foi especificado pelo componente pai. A checagem de tipos de `propTypes` acontece após `defaultProps` ser resolvido, logo também será aplicada ao `defaultProps`.

Reviewer 4: ```suggestion
A `defaultProps` será usadas para garantir que `this.props.name` tenha um valor caso não tenha sido especificado pelo componente pai. A checagem de tipos de `propTypes` acontece após `defaultProps` ser resolvida, logo a checagem também será aplicada à `defaultProps`.
```
Making just a little bit clearer.
